backend/src/main/java/gov/cdc/usds/simplereport/service/OrganizationService.java

```diff
@@ -51,12 +52,16 @@ public void resetOrganizationRolesContext() {
   }
 
   public Optional<OrganizationRoles> getCurrentOrganizationRoles() {
-    if (organizationRolesContext.hasBeenPopulated()) {
-      return organizationRolesContext.getOrganizationRoles();
+    try {
+      if (organizationRolesContext.hasBeenPopulated()) {
+        return organizationRolesContext.getOrganizationRoles();
+      }
+      var result = fetchCurrentOrganizationRoles();
+      organizationRolesContext.setOrganizationRoles(result);
+      return result;
+    } catch (ScopeNotActiveException e) {
+      return fetchCurrentOrganizationRoles();
```

Comment:
Can you remind me why we don't need to `organizationRolesContext.setOrganizationRoles(result);` here?

Reply:
Yep! `organizationRolesContext` at this point no longer exists - it only exists while the scope of the request is active, and at this point in our async method we no longer have a request scope to work with. So trying to `setOrganizationRoles` will result in a `NullPointerException`.